HADotNet.Core/BaseClient.cs

```diff
@@ -89,5 +91,31 @@ protected BaseClient(Uri instance, string apiKey)
 
             throw new Exception($"Unexpected response code {(int)resp.StatusCode} from Home Assistant API endpoint {path}.");
         }
+
+        /// <summary>
+        /// Performs a DELETE request on the specified path.
+        /// </summary>
+        /// <typeparam name="T">The type of data to deserialize and return.</typeparam>
+        /// <param name="path">The relative API endpoint path.</param>
+        /// <returns>The deserialized data of type <typeparamref name="T" />.</returns>
+        protected async Task<T> Delete<T>(string path) where T : class
+        {
+            var req = new RestRequest(path, Method.DELETE);
+
+            var resp = await Client.ExecuteTaskAsync(req);
+
+            if (!string.IsNullOrWhiteSpace(resp.Content) && (resp.StatusCode == HttpStatusCode.OK || resp.StatusCode == HttpStatusCode.Created))
+            {
+                // Weird case for strings - return as-is
+                if (typeof(T).IsAssignableFrom(typeof(string)))
+                {
+                    return resp.Content as T;
+                }
+
+                return JsonConvert.DeserializeObject<T>(resp.Content);
+            }
+
+            throw new Exception($"Unexpected response code {(int)resp.StatusCode} from Home Assistant API endpoint {path}.");
+        }
```

Comment:
If we have `Delete<T>`, it would also make sense to add `Delete` without a return type, since most `DELETE` operations return `204 No Content`. (May not be true for Home Assistant but you never know.)

Reply:
Jake, agree, added a new method with the suggested signature.